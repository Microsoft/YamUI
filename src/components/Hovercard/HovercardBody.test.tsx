/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { HovercardBody, HovercardBodyProps } from '.';

describe('<HovercardBody />', () => {
  let component: ShallowWrapper<HovercardBodyProps>;

  describe('with default options', () => {
    beforeEach(() => {
      component = shallow(<HovercardBody>body content</HovercardBody>);
    });

    it('renders its given content', () => {
      expect(component.render().text()).toEqual('body content');
    });

    it('contains its base className', () => {
      expect(component.hasClass('y-hovercard--body')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('with additional className', () => {
    beforeEach(() => {
      component = shallow(<HovercardBody className="TEST_CLASSNAME">body content</HovercardBody>);
    });

    it('includes that className', () => {
      expect(component.hasClass('TEST_CLASSNAME')).toBe(true);
    });

    it('still has its base className', () => {
      expect(component.hasClass('y-hovercard--body')).toBe(true);
    });

    it('matches its snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
