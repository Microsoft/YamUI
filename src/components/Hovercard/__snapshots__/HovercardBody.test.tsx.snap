// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HovercardBody /> with additional className matches its snapshot 1`] = `
<div
  className="y-hovercard--body TEST_CLASSNAME"
>
  <CustomizedBlock
    padding="xLarge"
  >
    body content
  </CustomizedBlock>
</div>
`;

exports[`<HovercardBody /> with default options matches its snapshot 1`] = `
<div
  className="y-hovercard--body"
>
  <CustomizedBlock
    padding="xLarge"
  >
    body content
  </CustomizedBlock>
</div>
`;
